### PR TITLE
go.mod: remove dependency on baremetal-operator

### PR DESCRIPTION
**Pull request by raghavendra-talur**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -172,8 +172,6 @@ require (
 replace k8s.io/client-go => k8s.io/client-go v0.26.4
 
 replace (
-	github.com/metal3-io/baremetal-operator/apis => github.com/openshift/baremetal-operator/apis v0.0.0-20211201170610-92ffa60c683d
-	github.com/metal3-io/baremetal-operator/pkg/hardwareutils => github.com/openshift/baremetal-operator/pkg/hardwareutils v0.0.0-20211201170610-92ffa60c683d
```
Comment (nirs) [inline go.mod:176]: Was it removed by `go mod tidy`? How these lines were added here?

Comment (raghavendra-talur) [reply go.mod:176]: If you note, these lines are in the replace section of go.mod and will not be automatically removed by `go mod tidy`. The go.mod and go.sum files don't have any other reference to baremetal-operator which tells me that we probably removed the dependency on that long back.

Comment (nirs) [reply go.mod:176]: Right, makes sense.

**Review by nirs — APPROVED** [review]
Looks good to me, I hope there is no hidden reason to have this replaces.

**Review by ShyamsundarR — APPROVED** [review]

**Pull request merged**